server/src/user/user.service.ts

```diff
@@ -926,31 +933,70 @@ export class UserService {
       communityId,
       ...rest
     } = body;
-    const isAdmin = await this.isAdmin(userId);
+
+    // Fetch the user only once
+    const user = await this.findOneOrFail(userId, ['leaderOf']);
```

Comment:
most of server code goodness is about optimizing database queries. this works totally fine and isnt a big deal since it wont get called a lot, but as an example of the type of optimization we want to do:

doing this loads the full community entities for the communities that this user is a leader of. we actually dont need that entity information at all: we only need the keys that are the ids of the communities that the user leads. so instead, we can use the `leaderOfIds` field on `user.entity.ts` and not need to load the extra entities